huaweicloud/services/nat/resource_huaweicloud_nat_dnat_rule.go

```diff
@@ -0,0 +1,306 @@
+package nat
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk"
+	"github.com/chnsz/golangsdk/openstack/nat/v2/dnats"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func ResourceDnatRule() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceDnatRuleCreate,
+		ReadContext:   resourceDnatRuleRead,
+		UpdateContext: resourceDnatRuleUpdate,
+		DeleteContext: resourceDnatRuleDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "The region where the DNAT rule is located.",
+			},
+			"floating_ip_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The ID of the floating IP address.",
+			},
+			"protocol": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The protocol type.",
+			},
+			"gateway_id": {
```

Comment:
is it necessary to replace nat_gateway_id with gateway_id?
if so, add some limit to avoid that they are both empty

Reply:
the API will throw error if they both empty.